spec/shared_print/updater_spec.rb

```diff
@@ -97,4 +96,80 @@
     arr = SharedPrint::Finder.new.commitments.to_a
     expect(arr.map(&:local_bib_id).sort).to eq ["b", "c", "d", "updated"]
   end
+
+  context "updating ocns" do
+    it "can update commitment ocn and move commitment to another cluster" do
+      cluster_tap_save [spc1, ht1, ht2]
+      expect(Cluster.find_by(ocns: ocn1).commitments.count).to eq 1
+      expect(Cluster.find_by(ocns: ocn2).commitments.count).to eq 0
+      updater.process_record(upd3)
+      expect(Cluster.find_by(ocns: ocn1).commitments.count).to eq 0
+      expect(Cluster.find_by(ocns: ocn2).commitments.count).to eq 1
+    end
+
+    it "moves to a new cluster if updating to the ocn of another cluster " do
+      # Start with a commitment on a cluster with ocns:[ocn1]
+      cluster_tap_save [ht1, ht2, spc1]
+
+      original_cluster_id = spc1.cluster._id
+      expect(spc1.cluster.ocns).to eq [spc1.ocn]
+
+      # If we update the commitment to ocn2
+      updater.process_record(upd3)
+      # ...the commitment should move to another cluster
+
+      new_cluster = Cluster.find_by(ocns: [ocn2])
+      expect(new_cluster.nil?).to be false
+      expect(new_cluster._id).not_to eq original_cluster_id
+    end
+
+    it "can update to an ocn on the same cluster" do
+      # set up a cluster with ocn1 and ocn2
+      resolution = build(:ocn_resolution, deprecated: ocn1, resolved: ocn2)
+      cluster = build(:cluster, ocns: [ocn1, ocn2])
+      cluster.save
+      cluster.add_ocn_resolutions(resolution)
+      expect(cluster.ocns).to eq [ocn1, ocn2]
+
+      # add an spc that has ocn1 (and a matching ht_item)
+      cluster_tap_save [spc1, ht1]
+      expect(Cluster.count).to eq 1
```

Comment:
I waffle a bit about these expectations -- on the one hand I think it's worthwhile as a kind of documentation for what the precondition is especially given the somewhat complex setup, but it also maybe muddies what the test is actually testing (lines 149-151). If nothing else, I think it would help to be more explicit with comments about what's setup, what's preconditions, and what the actual test is. It also might be worth refactoring some of the setup in these tests and in other similar tests - if we can find any common patterns or abstract the variation in setup some way it might in the end make the tests easier to understand. I think we have the

Reply:
https://hathitrust.atlassian.net/browse/DEV-546